rdr_service/genomic/genomic_job_components.py

```diff
@@ -1219,14 +1245,99 @@ def _get_new_biobank_samples(self, from_date):
             result = session.execute(_new_samples_sql, params).fetchall()
         return list(zip(*result))
 
+    def _get_new_c2_participants(self, from_date):
+        """
+        Retrieves C2 participants and validation data.
+        Broken out so that DNA samples' business logic is handled separately
+        :param from_date:
+        :return:
+        """
+        _c2_participant_sql = """
+            SELECT DISTINCT
+              ps.biobank_id,
+              ps.participant_id,
+              0 AS biobank_order_id,
+              0 AS collected_site_id,
+              0 AS biobank_stored_sample_id,
+              CASE
+                WHEN ps.withdrawal_status = :withdrawal_param THEN 1 ELSE 0
+              END as valid_withdrawal_status,
+              CASE
+                WHEN ps.suspension_status = :suspension_param THEN 1 ELSE 0
+              END as valid_suspension_status,
+              CASE
+                WHEN ps.consent_for_study_enrollment = :general_consent_param THEN 1 ELSE 0
+              END as general_consent_given,
+              CASE
+                WHEN ps.date_of_birth < DATE_SUB(now(), INTERVAL :dob_param*365 DAY) THEN 1 ELSE 0
```

Comment:
Finding the participant age this way doesn't account for leap years, `DATE_SUB(NOW(), INTERVAL :dob_param YEAR)` would.

Reply:
Good call, I will update.